apps/dav/lib/AppInfo/PluginManager.php

```diff
@@ -111,56 +114,70 @@ public function getAppPlugins() {
 	 * @return array
 	 */
 	public function getAppCollections() {
-		if (null === $this->collections) {
-			$this->populate();
-		}
+		$this->populate();
 		return $this->collections;
 	}
 
 	/**
 	 * @return IAddressBookProvider[]
 	 */
 	public function getAddressBookPlugins(): array {
-		if ($this->addressBookPlugins === null) {
-			$this->populate();
-		}
+		$this->populate();
 		return $this->addressBookPlugins;
 	}
 
 	/**
 	 * Returns an array of app-registered calendar plugins
 	 *
-	 * @return array
+	 * @return ICalendarProvider[]
 	 */
 	public function getCalendarPlugins():array {
-		if (null === $this->calendarPlugins) {
-			$this->populate();
-		}
+		$this->populate();
 		return $this->calendarPlugins;
 	}
 
 	/**
 	 * Retrieve plugin and collection list and populate attributes
 	 */
-	private function populate() {
-		$this->plugins = [];
-		$this->addressBookPlugins = [];
-		$this->calendarPlugins = [];
-		$this->collections = [];
+	private function populate(): void {
+		if ($this->populated) {
+			return;
+		}
+		$this->populated = true;
+
 		foreach ($this->appManager->getInstalledApps() as $app) {
 			// load plugins and collections from info.xml
 			$info = $this->appManager->getAppInfo($app);
 			if (!isset($info['types']) || !in_array('dav', $info['types'], true)) {
 				continue;
 			}
-			$this->loadSabrePluginsFromInfoXml($this->extractPluginList($info));
-			$this->loadSabreCollectionsFromInfoXml($this->extractCollectionList($info));
-			$this->loadSabreAddressBookPluginsFromInfoXml($this->extractAddressBookPluginList($info));
-			$this->loadSabreCalendarPluginsFromInfoXml($this->extractCalendarPluginList($info));
+			$plugins = $this->loadSabrePluginsFromInfoXml($this->extractPluginList($info));
```

Comment:
```suggestion
			$this->plugins = $this->loadSabrePluginsFromInfoXml($this->extractPluginList($info));
```

maybe? I don't see a second write to $this->plugins.

Reply:
It loops ;) so that would overwite it